SerialPrograms/Source/PokemonSV/Programs/PokemonSV_Terarium.cpp

```diff
@@ -0,0 +1,383 @@
+#include "CommonFramework/InferenceInfra/InferenceRoutines.h"
+#include "CommonFramework/Tools/ErrorDumper.h"
+#include "CommonFramework/OCR/OCR_NumberReader.h"
+#include "CommonFramework/ImageTypes/ImageViewRGB32.h"
+#include "CommonFramework/Exceptions/ProgramFinishedException.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_Superscalar.h"
+#include "NintendoSwitch/NintendoSwitch_Settings.h"
+#include "NintendoSwitch/Programs/NintendoSwitch_GameEntry.h"
+#include "PokemonSV/Programs/PokemonSV_GameEntry.h"
+#include "PokemonSV/Programs/PokemonSV_SaveGame.h"
+#include "PokemonSwSh/Commands/PokemonSwSh_Commands_DateSpam.h"
+#include "PokemonSwSh/Programs/PokemonSwSh_GameEntry.h"
+#include "PokemonSV/PokemonSV_Settings.h"
+#include "PokemonSV/Programs/PokemonSV_GameEntry.h"
+#include "PokemonSV/Inference/Dialogs/PokemonSV_DialogDetector.h"
+#include "PokemonSV/Inference/Battles/PokemonSV_NormalBattleMenus.h"
+#include "PokemonSV/Inference/Battles/PokemonSV_EncounterWatcher.h"
+#include "PokemonSV/Inference/PokemonSV_MainMenuDetector.h"
+#include "PokemonSV/Inference/Dialogs/PokemonSV_DialogDetector.h"
+#include "PokemonSV/Inference/Overworld/PokemonSV_OverworldDetector.h"
+#include "PokemonSV/Programs/PokemonSV_Navigation.h"
+#include "PokemonSwSh/Commands/PokemonSwSh_Commands_DateSpam.h"
+#include "PokemonSV/Inference/PokemonSV_BlueberryQuestDetector.h"
+#include "PokemonSV/Programs/Battles/PokemonSV_BasicCatcher.h"
+#include "PokemonSV_Terarium.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+void return_to_plaza(const ProgramInfo& info, ConsoleHandle& console, BotBaseContext& context) {
+    console.log("Attempting to return to Central Plaza.");
+    //Modified version of handle_battles_and_back_to_pokecenter()
+    bool returned_to_pokecenter = false;
+    bool laggy = false;
+
+    while(!returned_to_pokecenter){
+        EncounterWatcher encounter_watcher(console, COLOR_RED);
+        int ret = run_until(
+            console, context,
+            [&](BotBaseContext& context){
+                //Exit any dialogs (ex. Cyrano upgrading BBQs)
+                OverworldWatcher overworld(COLOR_RED);
+                int ret_overworld = run_until(
+                    console, context,
+                    [&](BotBaseContext& context) {
+                        pbf_mash_button(context, BUTTON_B, 10000);
+                    },
+                    { overworld }
+                    );
+                if (ret_overworld == 0) {
+                    console.log("Overworld detected.");
+                }
+                context.wait_for_all_requests();
+
+                try {
+                    open_map_from_overworld(info, console, context);
+                }
+                catch (...) {
+                    //Stuck climbing on a wall? Center camera and try to backwards jump off.
```

Comment:
The idea of getting off the wall is interesting since we have this problem elsewhere. (platform bot has this problem)

Here you're only retrying once. But from my experience that may not be enough depending on how you get stuck. So what I'm thinking is that we could pull this out in a generic "get off the wall" function with multiple retries before finally giving up.

Here you're using the `open_map_from_overworld()` to determine if you've successfully jumped off the wall. And I think in most of our other cases it's the same thing.

Exactly how we want to do this I'm not sure. Could either add this retry capability to `open_map_from_overworld()` itself or define a new function that wraps it.